src/main/java/org/jenkinsci/plugins/docker/workflow/WithContainerStep.java

```diff
@@ -177,6 +177,12 @@ private static void destroy(String container, Launcher launcher, Node node, EnvV
             }
 
             container = dockerClient.run(env, step.image, step.args, ws, volumes, volumesFromContainers, envReduced, dockerClient.whoAmI(), /* expected to hang until killed */ "cat");
+            final List<String> ps = dockerClient.listProcess(env, container);
+            if (!ps.contains("cat")) {
+                listener.error("The container started but didn't ran the expected command. Please double check your Entrypoint does execute the command passed as docker run argument. See https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#entrypoint for entrypoint best practices.");
+                throw new IOException("Failed to run container with CMD `cat`");
```

Comment:
Probably better to print the warning but let it continue, in case the user knows something we do not.

Reply:
I was not sure about this, good point